src/converse-register.js

```diff
@@ -390,6 +417,14 @@
                     if (ev && ev.preventDefault) { ev.preventDefault(); }
                     _converse.connection.reset();
                     this.render();
+                    if (_converse.registration_domain) {
+                        this.renderRegistrationRequest(__('Retry'));
+                        var _register_panel = this;
+                        document.querySelector('button.button-cancel').onclick =  
+                            function () { 
+                                _register_panel.fetchRegistrationForm(_converse.registration_domain); 
+                            };
```

Comment:
We can use partial application here instead of writing out a function.

Try this:
```
document.querySelector('button.button-cancel').onclick =  
      _.bind(this, this.fetchRegistrationForm, _converse.registration_domain);
```